gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufdataset.cpp

```diff
@@ -0,0 +1,302 @@
+/************************************************************************/
+
+OGRFlatGeobufDataset::OGRFlatGeobufDataset()
+{
+
+}
+
+OGRFlatGeobufDataset::OGRFlatGeobufDataset(const char *osDirName)
+{
+    CPLDebug("FlatGeobuf", "Request to create dataset %s", osDirName);
+    m_create = true;
+    m_pszName = CPLStrdup(osDirName);
+}
+
+/************************************************************************/
+/*                         ~OGRFlatGeobufDataset()                          */
+/************************************************************************/
+
+OGRFlatGeobufDataset::~OGRFlatGeobufDataset()
+{
+}
+
+/************************************************************************/
+/*                                Open()                                */
+/************************************************************************/
+
+GDALDataset *OGRFlatGeobufDataset::Open(GDALOpenInfo* poOpenInfo)
+{
+    if( !OGRFlatGeobufDriverIdentify(poOpenInfo) || poOpenInfo->eAccess == GA_Update )
+        return nullptr;
+
+    VSILFILE *fp = poOpenInfo->fpL;
+    CPLString osFilename(poOpenInfo->pszFilename);
+
+    uint64_t offset = sizeof(magicbytes);
+    CPLDebug("FlatGeobuf", "Start at offset (%lu)", sizeof(magicbytes));
+    VSIFSeekL(fp, offset, SEEK_SET);
+    uint32_t headerSize;
+    VSIFReadL(&headerSize, 4, 1, fp);
+    CPLDebug("FlatGeobuf", "headerSize (%d)", headerSize);
+    GByte* buf = static_cast<GByte*>(VSI_MALLOC_VERBOSE(headerSize));
+    VSIFReadL(buf, 1, headerSize, fp);
+    auto header = GetHeader(buf);
+    offset += 4 + headerSize;
+    CPLDebug("FlatGeobuf", "Add headerSize to offset (%d)", 4 + headerSize);
+
+    auto featuresCount = header->features_count();
+    auto index_node_size = header->index_node_size();
+    if (index_node_size > 0) {
+        auto treeSize = PackedRTree::size(featuresCount);
+        offset += treeSize;
+        CPLDebug("FlatGeobuf", "Add treeSize to offset (%zu)", treeSize);
+        offset += featuresCount * 8;
+        CPLDebug("FlatGeobuf", "Add featuresCount * 8 to offset (%zu)", featuresCount * 8);
+    }
+
+    CPLDebug("FlatGeobuf", "Features start at offset (%zu)", offset);
+
+    auto poDS = new OGRFlatGeobufDataset();
+    poDS->SetDescription(osFilename);
+
+    poDS->m_apoLayers.push_back(
+        std::unique_ptr<OGRLayer>(new OGRFlatGeobufLayer(header, osFilename, offset))
+    );
+
+    return poDS;
+}
+
+GDALDataset *OGRFlatGeobufDataset::Create( const char *pszName,
+                                        CPL_UNUSED int nBands,
+                                        CPL_UNUSED int nXSize,
+                                        CPL_UNUSED int nYSize,
+                                        CPL_UNUSED GDALDataType eDT,
+                                        char ** /* papszOptions */)
+{
+    // First, ensure there isn't any such file yet.
+    VSIStatBufL sStatBuf;
+
+    if (strcmp(pszName, "/dev/stdout") == 0)
```

Comment:
does streaming to /dev/stdout really work ?

Reply:
Not sure but would like it to :) Need to figure out a test for that I guess.

Reply:
Removed for now.